server/src/app/GaWorker.ts

```diff
@@ -113,4 +113,58 @@ export class GaWorker {
 
     return startSet.map((s: string) => new Note(s));
   }
+
+  // Function to assign weights to possible notes for mutation
+  private assignNoteWeights(index: number, mut_notes: Array<Note>, melody: Melody): Array<number> {
```

Comment:
Also maybe lets move this to a utils function file. The GA worker is quite busy.